### PR TITLE
Add OpenAPI icon addition tool

### PR DESCRIPTION
**Pull request by kapilraajP**

### CODE REVIEW

File: resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Maven Central repository" />
+      <option name="url" value="https://repo1.maven.org/maven2" />
+    </remote-repository>
+    <remote-repository>
```
Comment (abeykoon) [inline resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml:9]: Why do we need this repository? Do we pull something specific from JBoss?

File: resources/scripts/openapi-icon-tool/.idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (abeykoon) [inline resources/scripts/openapi-icon-tool/.idea/misc.xml:1]: Wha does this file do?

File: resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml (hunk 1 above)
Comment (abeykoon) [reply resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml:9]: Actually we use `Gradle` right? Do we need this file at all?

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.io.IOException;
+import java.util.List;
+
+public class IconTool {
+    static String iconsSourcePath;
+    static String openApiConnectorsPath;
+    public static void main(String[] args) throws IOException {
+        iconsSourcePath = Utils.getConfigValue("iconsSourcePath");
+        openApiConnectorsPath = Utils.getConfigValue("openApiConnectorsPath");
+        List<String> ballerinaPackages = Utils.findBallerinaPackages(openApiConnectorsPath);
```
Comment (abeykoon) [inline resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:11]: We need to update the icon of all connectors to what we have in `iconsSourcePath ` with this tool I guess. Does this util function return all the connector packages we have or only changed ones? 

Or does this tool is updating icons of changed connectors only?

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        for (String packagePath : ballerinaPackages) {
+            String[] packageNameSplit = packagePath.split("/");
+            String moduleName = packageNameSplit[packageNameSplit.length - 1];
+            boolean resourcesFolderExist = Utils.isResourcesFolderExist(packagePath);
+            if (resourcesFolderExist) {
+                String iconPath = Utils.findConnectorIcon(moduleName, iconsSourcePath);
+                if (iconPath != null) {
+                    Utils.addReplaceIcon(packagePath, iconPath, moduleName);
+                } else {
+                    System.out.println("No icon available for " + packagePath);
```
[the remark this answers is not in the document]
Comment (abeykoon) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:21]: Usually we import commons logger from Apache and do it. 
https://mvnrepository.com/artifact/commons-logging/commons-logging

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                }
+            } else {
```
Comment (abeykoon) [inline resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:23]: This if and else has common stuff. What we do additionally is creating resources folder. What we can do is

```
createResoucesFolderIfNotExists(path or whatever params);
addNewIcon(packagePath, iconPath, moduleName);

private void  addNewIcon (String packagePath, String iconPath, String moduleName) {
  if (iconPath != null) {
                      Utils.addReplaceIcon(packagePath, iconPath, moduleName);
  } else {
                      System.out.println("No icon available for " + packagePath);
  }
}
```

Comment (abeykoon) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:23]: WDYT?

**Comment by abeykoon** [comment]
Shall we add WSO2 licence heading to all files just in case? You can configure in IDE to automatically add it to any file.

File: resources/scripts/openapi-icon-tool/.idea/.gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
[the remark this answers is not in the document]
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/.idea/.gitignore:1]: Removed .idea files

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:21]: Ack. Changed this to log using `org.apache.log4j.Logger`

File: resources/scripts/openapi-icon-tool/.idea/misc.xml (hunk 2 above)
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/.idea/misc.xml:1]: This files are auto added one. These are removed while removing `.idea` folder

File: resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml (hunk 1 above)
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/.idea/jarRepositories.xml:9]: This files are auto added one. These are removed while removing `.idea` folder

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java (hunk 3 above)
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:11]: It updates all the icons

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java (hunk 5 above)
Comment (kapilraajP) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:35]: Ack

File: resources/scripts/openapi-icon-tool/src/main/java/IconTool.java (hunk 3 above)
Comment (abeykoon) [reply resources/scripts/openapi-icon-tool/src/main/java/IconTool.java:27]: +1. Then that would be fine.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**